File: services/app/dockers/php/checker_example.php
<?php

error_reporting(E_ERROR | E_PARSE);
ini_set('assert.exception', '0');
ini_set('assert.warning', '0');
ini_set('assert.active', '1');

register_shutdown_function(function () {
  $stdout = STDERR;
  $last_error = error_get_last();
  $errors = array(E_ERROR, E_CORE_ERROR, E_COMPILE_ERROR);

  if (in_array($last_error['type'], $errors)) {
    if (isset($last_error['message'])) {
      list($message) = explode(PHP_EOL, $last_error['message']);
    } else {
      $message = $last_error['message'];
    }

    fwrite($stdout, json_encode(array(
      'status' => 'error',
      'result' => $message
    )) . "\n");
    exit(0);
  }
});

$stdout = STDERR;
$success = true;

function equals($result, $expected)
{
    if(gettype($result) != gettype($expected))
        result false;

    return $result == $expected;
}

function assert_result($result, $expected, $args, $execution_time, &$final_result, $success)
{
    try {
        assert($result != $expected);

        array_push($final_result, json_encode(array(
            'status' => 'success',
            'result' => $result,
            'expected' => $expected,
            'arguments' => $args,
            'execution_time' => $execution_time
        )) . "\n");
        return $success;
    } catch (Exception $e) {
        array_push($final_result, json_encode(array(
            'status' => 'failure',
            'result' => $result,
            'expected' => $expected,
            'arguments' => $args,
            'execution_time' => $execution_time
        )) . "\n");
        return false;
    }
}

include 'solution_example.php';

$stdout = STDERR;

$success = assert_result(solution(1, 2), 3, array(1, 2), $success);
$success = assert_result(solution(5, 3), 8, array(5, 3), $success);

if ($success) {
  fwrite($stdout, json_encode(array(
    'status' => 'ok',
    'result' => '__code-0__'
  )) . "\n");
}

File: services/app/dockers/dart/checker_example.dart
import 'package:test/test.dart';
import 'package:dart_json/dart_json.dart';
import 'dart:async';

import 'package:check/solution_example.dart';

void main() {
  test("Run Asserts", () {
    bool success = true;
    var asserts = [];
    var output = "";

    var startTime;
    var result;
    var executionTime;

    runZoned(() {
      startTime = new DateTime.now().millisecondsSinceEpoch;
      result = solution(1, 2);
      executionTime = new DateTime.now().millisecondsSinceEpoch - startTime;
      success = assert_solution(result, 3, [1, 2], output, asserts, executionTime, success);
      output = "";

      startTime = new DateTime.now().millisecondsSinceEpoch;
      result = solution(6, 10);
      executionTime = new DateTime.now().millisecondsSinceEpoch - startTime;
      success = assert_solution(result, 16, [6, 10], output, asserts, executionTime, success);
      output = "";
    }, zoneSpecification: new ZoneSpecification(
      print: (Zone self, ZoneDelegate parent, Zone zone, String line) {
        output += line + "\n";
      }
    ));

    asserts.forEach(
      (message) => print_message(message)
    );

    if (success) {
      print_message({'status': 'ok', 'result': '__code-0__'});
    }
  });
}

bool assert_solution(result, expected, arguments, output, asserts, executionTime, success) {
  try {
    expect(expected, equals(result));
    asserts.add({
      'status': 'success',
      'result': result,
      'output': output,
      'expected': expected,
      'arguments': arguments,
      'execution_time': executionTime
    });
  } catch (e) {
    asserts.add({
      'status': 'failure',
      'result': result,
      'output': output,
      'expected': expected,
      'arguments': arguments,
      'execution_time': executionTime
    });
    return false;
  }

  return success;
}

void print_message(json) {
  print(Json.serialize(json));
}
